File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/07_Moving.js
function moving(input) {

    // 4.	На следващите редове (до получаване на команда "Done") - брой кашони, които се пренасят в квартирата - цели числа в интервала [1...10000];
    // Функцията трябва да приключи прочитането на данни при команда "Done" или ако свободното място свърши.

    let width = Number(input[0]);
    let length = Number(input[1]);
    let higth = Number(input[2]);

    let totalSpace = width * length * higth;
    let freeSpace = totalSpace;

    let i = 3;
    while (true) {
        let currentCommand = input[i];
        i++;

        if (currentCommand == "Done") {
            console.log(`${freeSpace} Cubic meters left.`);
            break;
        }

        let currentBoxes = Number(currentCommand);
        freeSpace -= currentBoxes;

        if (freeSpace < 0) {
            console.log(`No more free space! You need ${-freeSpace} Cubic meters more.`);  // Math.abs(freeSpace)
            break;
        }

    }

}


moving(["10",
    "10",
    "2",
    "20",
    "20",
    "20",
    "20",
    "122"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/05_While_Loop_Exercise/07_Moving.py
width = int(input())
length = int(input())
height = int(input())

total_space = width * length * height

# 4.	На следващите редове (до получаване на команда "Done") - брой кашони, които се пренасят в квартирата - цели числа
command = input()  # 'Done' or spaces as a str (e.g. '10')
while command != 'Done':
    spaces = int(command)

    total_space -= spaces
    if total_space <= 0:
        break

    command = input()

if total_space >= 0:
    print(f'{total_space} Cubic meters left.')
else:
    print(f'No more free space! You need {-total_space} Cubic meters more.')
